llvm/lib/Transforms/Scalar/LoopIdiomRecognize.cpp

```diff
@@ -1545,6 +1545,12 @@ bool LoopIdiomRecognize::optimizeCRCLoop(const PolynomialInfo &Info) {
   if (TT.getArch() == Triple::hexagon)
     return false;
 
+  // Check that the DataLayout allows us to index into the new 256-entry CRC
+  // table.
+  Type *IdxTy = SE->getDataLayout().getIndexType(M.getContext(), 0);
+  if (IdxTy->getIntegerBitWidth() < 8)
+    return false;
```

Comment:
I'm not sure if there's a better way to do this?

Reply:
I don't think we need this is `if`. In this very exotic scenario the ZExt would assert.

Reply:
Hm, pending on @nikic's expertise, I don't think anything in LLVM would work in the exotic scenario.